### DISCUSSION TITLE
TPS calculation

### DISCUSSION THREAD

**Discussion by BhuiyanSAP** [artilleryio/artillery#2229] [Artillery]
1. How to do the TPS calculation of test result?
2. What is the background calculation of  **aggregate** **"http.request_rate": 46** ?
 `  "aggregate": {
    "counters": {
      "vusers.created_by_name.Post Order": 6120,
      "vusers.created": 6120,
      "http.requests": 6120,
      "http.codes.201": 6120,
      "http.responses": 6120,
      "http.downloaded_bytes": 403920,
      "plugins.metrics-by-endpoint.postOrder.codes.201": 6120,
      "vusers.failed": 0,
      "vusers.completed": 6120
    },
    "rates": {
      "http.request_rate": 46
    },`

**Comment by hassy** [comment]
hi @nazmuncintas 👋 how do you define TPS in your system? Does "transaction" = "HTTP request"? if that's the case then TPS is provided by the `http.request_rate` metric. The calculation for that metric is `total number of HTTP requests in 10 seconds / 10`, so it's the average over a 10s window.

**Reply by BhuiyanSAP** [reply]
Hi @hassy Thanks a lot. So in the report.json file the value of **http.request_rate** (inside the **aggregate** tag) is the average over a 10s window?

**Reply by hassy** [reply]
yes that's correct @nazmuncintas

**Reply by BhuiyanSAP** [reply]
Hi @hassy One more question, we run our test for 2mins, but we found the duration is more then 2mins in the html report. Example as follows:
![image](https://github.com/artilleryio/artillery/assets/26798364/71562c97-f821-4e2b-ac70-89134078d3e4)

The overall **TPS** calculation will be **6840(request)/121 (duration)** or **6840(request)/120 (test setup time)**

**Reply by hassy** [reply]
@nazmuncintas the TPS reported by Artillery (via `http.request_rate` metric) takes only the actual execution time of the test into account (and not any setup/teardown time) so the accuracy as reported by Artillery is not affected
